pkg/sync/map.go

```diff
@@ -0,0 +1,171 @@
+package sync
+
+import (
+	"sync"
+
+	"golang.org/x/exp/maps"
+)
+
+// Map is like a Go map[interface{}]interface{} but is safe for concurrent use by multiple goroutines.
+type Map[K comparable, V any] struct {
+	mutex sync.RWMutex
+	data  map[K]V
+}
+
+// NewMap creates map.
+func NewMap[K comparable, V any]() *Map[K, V] {
+	return &Map[K, V]{
+		data: make(map[K]V),
+	}
+}
+
+// Store sets the value for a key.
+func (m *Map[K, V]) Store(key K, value V) {
+	m.StoreWithFunc(key, func() V { return value })
+}
+
+func (m *Map[K, V]) StoreWithFunc(key K, createFunc func() V) {
```

Comment:
I don't see usability of this function.

Reply:
I've copied the interface from https://github.com/plgd-dev/kit/blob/main/sync/map.go, but yeah in general the 'withFunc' methods don't seem all that useful.